subprojects/core/src/main/java/org/gradle/initialization/buildsrc/BuildSourceBuilder.java

```diff
@@ -93,14 +105,29 @@ public String getBuildPath() {
     }
 
     private ClassPath buildBuildSrc(StartParameter startParameter) {
-        BuildController buildController = createBuildController(startParameter);
+        final PersistentCache buildSrcCache = createCache(startParameter);
```

Comment:
If we only use this as a lock, should we reflect that in the types being used? What we are creating here is a closable lock, not a cache. I wonder how much refactor that would require.